pkg/controller/service/capture.go

```diff
@@ -48,6 +53,28 @@ type Node struct {
 	Labels map[string]string `json:"labels"`
 }
 
+func (c *controller) InitInformer() {
+	if c.k8sClient == nil {
+		return
+	}
+
+	c.k8sInformerFactory = informers.NewSharedInformerFactory(c.k8sClient, time.Minute*1)
+	c.podInformer = c.k8sInformerFactory.Core().V1().Pods()
+	c.podInformer.Informer().GetIndexer().AddIndexers(cache.Indexers{
+		"nodeName": func(obj interface{}) ([]string, error) {
+			pod, ok := obj.(*podv1.Pod)
+			if !ok {
+				return []string{}, nil
+			}
+			// We are only interested in active pods with nodeName set
+			if len(pod.Spec.NodeName) == 0 || pod.Status.Phase == podv1.PodSucceeded || pod.Status.Phase == podv1.PodFailed {
```

Comment:
May we only need to see if `pod.Status.Phase` is `Running`

```
   Possible enum values:
     - `"Failed"` means that all containers in the pod have terminated, and at
    least one container has terminated in a failure (exited with a non-zero exit
    code or was stopped by the system).
     - `"Pending"` means the pod has been accepted by the system, but one or
    more of the containers has not been started. This includes time before being
    bound to a node, as well as time spent pulling images onto the host.
     - `"Running"` means the pod has been bound to a node and all of the
    containers have been started. At least one container is still running or is
    in the process of being restarted.
     - `"Succeeded"` means that all containers in the pod have voluntarily
    terminated with a container exit code of 0, and the system is not going to
    restart any of these containers.
     - `"Unknown"` means that for some reason the state of the pod could not be
    obtained, typically due to an error in communicating with the host of the
    pod. Deprecated: It isn't being set since 2015
    (74da3b14b0c0f658b3bb8d2def5094686d0e9095)
```